backend/typescript/services/implementations/camperService.ts

```diff
@@ -350,89 +351,120 @@ class CamperService implements ICamperService {
-          waitlistedCamper.campSession,
-          {
-            $push: { waitlist: newWaitlistedCamper.id },
-          },
-          { runValidators: true },
+      // Ensure all sessions were found
+      if (sessionsToWaitlist.length !== campSessions.length) {
+        throw new Error(
+          "The session ids passed in did not match the sessions found",
         );
+      }
 
-        if (!existingCampSession) {
-          throw new Error(
-            `Camp session ${waitlistedCamper.campSession} not found.`,
-          );
-        }
-
-        const camp = await MgCamp.findById(existingCampSession.camp);
-        if (!camp) {
-          throw new Error(`Camp ${existingCampSession.camp} not found.`);
-        }
-        await emailService.sendParentWaitlistConfirmationEmail(
-          camp,
-          existingCampSession,
-          [newWaitlistedCamper],
+      const findCamp = await MgCamp.findById(
+        sessionsToWaitlist[0].camp,
+        {},
+        { session },
+      );
+      if (!findCamp) {
+        throw new Error(
+          "Unable to find the camp associated with the camp sessions passed in",
         );
-      } catch (mongoDbError: unknown) {
-        // rollback user creation
-        try {
-          const deletedWaitlistedCamper: WaitlistedCamper | null = await MgWaitlistedCamper.findByIdAndDelete(
-            newWaitlistedCamper.id,
+      }
+      camp = findCamp;
+
+      // Ensure all camper(s) meet the age requirements of the camp
+      if (
+        !waitlistedCampers.every(
+          (c) => c.age >= camp.ageLower && c.age <= camp.ageUpper,
+        )
+      ) {
+        throw new Error(
+          "Some camper(s) didn't meet the age requirements of the camp",
+        );
+      }
+
+      // Create a waitlist camper entity for each session
+      const campersToWaitlist: Array<Omit<WaitlistedCamperDTO, "id">> = [];
+      campSessions.forEach((cs) => {
+        waitlistedCampers.forEach((waitlistedCamper) => {
+          const camperToWaitlist: Omit<WaitlistedCamperDTO, "id"> = {
+            firstName: waitlistedCamper.firstName,
+            lastName: waitlistedCamper.lastName,
+            age: waitlistedCamper.age,
+            contactName: waitlistedCamper.contactName,
+            contactEmail: waitlistedCamper.contactEmail,
+            contactNumber: waitlistedCamper.contactNumber,
+            campSession: cs,
+            status: "NotRegistered",
+          };
+          campersToWaitlist.push(camperToWaitlist);
+        });
+      });
+      // Insert the waitlisted camper entities
+      newWaitlistedCampers = await MgWaitlistedCamper.insertMany(
+        campersToWaitlist,
+        { session },
+      );
+
+      // Add the campers to the waitlist field of each session
+      await Promise.all(
+        sessionsToWaitlist.map((cs) => {
+          cs.waitlist.push(
+            ...newWaitlistedCampers.filter(
+              (camper) => camper.campSession.toString() === cs.id,
```

Comment:
might be understanding wrong, when would this condition fail? won't all campers belong to all the `sessionsToWaitlist`?